### PR TITLE
Add support for query params sharing

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
## Purpose
Resolves 
- https://github.com/wso2-enterprise/asgardeo-product/issues/9936

### CODE REVIEW

File: io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java [hunk 1]
```diff
@@ -65,6 +65,9 @@ public void sendForLogin(HttpServletRequest request, HttpServletResponse respons
         RequestContext requestContext = defaultOIDCManager.sendForLogin(request, response);
         if (request.getRequestURI() != null) {
             String redirectPageURI = request.getRequestURI().substring(request.getContextPath().length() + 1);
+            if (request.getQueryString() != null) {
+                redirectPageURI = redirectPageURI + "?" + request.getQueryString();
```
Comment (chamathns) [inline io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java:69]: let's use a StringBuilder to build the redirect URI

File: io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java [hunk 2]
```diff
@@ -65,6 +65,9 @@ public void sendForLogin(HttpServletRequest request, HttpServletResponse respons
         RequestContext requestContext = defaultOIDCManager.sendForLogin(request, response);
         if (request.getRequestURI() != null) {
             String redirectPageURI = request.getRequestURI().substring(request.getContextPath().length() + 1);
+            if (request.getQueryString() != null) {
```
[the remark this answers is not in the document]
Comment (ChanikaRuchini) [reply io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java]: Fixed

File: io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java (hunk 1 above)
Comment (ChanikaRuchini) [reply io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java]: Fixed

File: io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java [hunk 3]
```diff
@@ -64,8 +65,13 @@ public void sendForLogin(HttpServletRequest request, HttpServletResponse respons
         HttpSession session = request.getSession();
         RequestContext requestContext = defaultOIDCManager.sendForLogin(request, response);
         if (request.getRequestURI() != null) {
-            String redirectPageURI = request.getRequestURI().substring(request.getContextPath().length() + 1);
-            requestContext.setParameter(SSOAgentConstants.REDIRECT_URI_KEY, redirectPageURI);
+            StringBuilder redirectPageURI=new StringBuilder();
```
Comment (chamathns) [inline io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java:68]: fix formatting

File: io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java [hunk 4]
```diff
@@ -64,8 +65,13 @@ public void sendForLogin(HttpServletRequest request, HttpServletResponse respons
...
+            redirectPageURI.append(request.getRequestURI().substring(request.getContextPath().length() + 1));
+            if (StringUtils.isNotBlank(request.getQueryString())) {
+                redirectPageURI.append("?");
+                redirectPageURI.append(request.getQueryString());
```
Comment (chamathns) [inline io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java:72]:
```suggestion
                redirectPageURI.append("?").append(request.getQueryString());
```
let's combine these two

File: io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java (hunk 3 above)
Comment (ChanikaRuchini) [reply io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java]: Fixed

File: io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java (hunk 4 above)
Comment (ChanikaRuchini) [reply io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java]: Fixed

**Review by chamathns — APPROVED** [review]

**Pull request merged**